### DISCUSSION TITLE
Ideas and Feature Requests

### DISCUSSION THREAD

**Discussion by kashifkhan0771** [kashifkhan0771/utils#16] [Ideas]
Feel free to drop suggestions on:

- Useful utility functions you’d like to see
- Common pain points the library could address
- Improvements or optimizations to existing code

Looking forward to your input!

Thanks!

**Comment by shahzadhaider1** [comment]
Cool. Let's rock and rollll

**Comment by kashifkhan0771** [comment]
I am thinking about improving readMe. Maybe we can only mention available function names and what they do and add examples somewhere separate. Thoughts? Else the readME might become too hard to read :(

**Reply by shahzadhaider1** [reply]
Yes, good suggestion. As we add more and more utility functions, it'll be difficult to add all of them to readMe. 
We can create a new directory `/examples` and, inside that, we can create separate directories for each package, which can include detailed examples.

**Reply by kashifkhan0771** [reply]
I created an issue for this. I am more inclined towards creating a EXAMPLES.md file instead of putting examples in a package (working examples). I don't think that would be required.

**Reply by kashifkhan0771** [reply]
Head's up for `v0.8.0`, I am adding this. Not accepting any other PR's for next release until this is done.

**Comment by kashifkhan0771** [comment]
How about a function that takes a string and characters min limit and generate a random string of more than limit that contains that string for testing purposes. Another good thing can be is to give option to generate string in different formats like JSON, XML, HTML etc.

Example:
```
testStr, err := generateRandomString("this should be added", 100, "JSON")
if err != nil{
    fmt.Errorf("%v", err)
}

fmt.Println(testStr)
/*
Example output:
{
"test": "yjaurasasaesas"
"somerandom": "somerandomcharthatdoesn'tmakesense"
"tyyss": "this should be added" <- the passed string is added in returned random string
}
*/
testStr2, err := generateRandomString("this should be added", 100, "HTML")
if err != nil{
    fmt.Errorf("%v", err)
}

fmt.Println(testStr)
/*
Example output:
<html>
<head>random words</head>
<body>
tyyss: "this should be added" <- the passed string is added in returned random string
<p>nothing just more random words</p>
</body>
</html>
*/

I know this can be challenging, but this can be used for testings where we need a random string for testing that must have some words.

**Reply by shahzadhaider1** [reply]
where can we use that? I am asking because the application of this might be minimal

**Reply by kashifkhan0771** [reply]
The usecase for this is mostly in testcases where you need some sort of response back in plain text, JSON or any other format. One example on my mind is that where you are mocking response from a server and you need a JSON with some specific message for test cases.

**Comment by kashifkhan0771** [comment]
I am thinking about adding some file type conversion or you can say migration features e.g. Text to PDF or CSV to JSON etc. It can be complex but a good feature. Although it will take time to finalize all this but this is in my mind.

**Comment by shahzadhaider1** [comment]
that is a major thing but could be really helpful. I recently needed it (HTML to PDF) and struggled to find a simple solution.

**Comment by kashifkhan0771** [comment]
The `README` and `EXAMPLES` file is really growing. Any ideas or suggestion how can we handle these more better?

**Reply by kashifkhan0771** [reply]
Issue: https://github.com/kashifkhan0771/utils/issues/103

**Comment by kashifkhan0771** [comment]
Closing this discussion as I opened an issue similar for this: https://github.com/kashifkhan0771/utils/issues/110

**Reply by kashifkhan0771** [reply]
Add your ideas or feature requests in that issue for discussion.
